quickstart/201-azfw-with-secure-hub/main.tf

```diff
@@ -0,0 +1,370 @@
+  depends_on = [
+    azurerm_resource_group.azfw_rg
+  ]
+}
+
+// Create a Azure Firewall Policy Rule Collection Group
+// Create a Application Rule Collection
+// Create rules for Windows Update
+// Create rules for Microsoft.com
+resource "azurerm_firewall_policy_rule_collection_group" "app_policy_rule_collection_group" {
+  name               = "DefaulApplicationtRuleCollectionGroup"
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  priority           = 300
+  application_rule_collection {
+    name     = "DefaultApplicationRuleCollection"
+    action   = "Allow"
+    priority = 100
+    rule {
+      name        = "Allow-MSFT"
+      description = "Allow access to Microsoft.com"
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      destination_fqdns = ["*.microsoft.com"]
+      terminate_tls     = false
+      source_addresses  = ["*"]
+    }
+  }
+  depends_on = [
+    azurerm_firewall_policy.azfw_policy
+  ]
+}
+
+// Create the Azure Firewall
+resource "azurerm_firewall" "fw" {
+  name                = "fw-azfw-securehub-eus"
+  location            = azurerm_resource_group.azfw_rg.location
+  resource_group_name = azurerm_resource_group.azfw_rg.name
+  sku_name            = "AZFW_Hub"
+  sku_tier            = var.fw_sku
+  tags                = azurerm_resource_group.azfw_rg.tags
+  virtual_hub {
+    virtual_hub_id  = azurerm_virtual_hub.azfw_vwan_hub.id
+    public_ip_count = 1
+  }
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  depends_on = [
+    azurerm_firewall_policy.azfw_policy,
+    azurerm_virtual_hub.azfw_vwan_hub
+  ]
+}
+
+// Create Virtual Network, Subnets, PIP, NICs, NSGs, and NIC-NSG associations
+// Create a Virtual Network
+resource "azurerm_virtual_network" "azfw_vnet" {
+  name                = "vnet-azfw-securehub-eus"
+  location            = azurerm_resource_group.azfw_rg.location
+  resource_group_name = azurerm_resource_group.azfw_rg.name
+  address_space       = ["10.10.0.0/16"]
+  tags                = azurerm_resource_group.azfw_rg.tags
+  depends_on = [
+    azurerm_resource_group.azfw_rg
+  ]
+}
+
+// Create a Subnet for Workload VMs
+resource "azurerm_subnet" "workload_subnet" {
+  name                 = "subnet-workload"
+  resource_group_name  = azurerm_resource_group.azfw_rg.name
+  virtual_network_name = azurerm_virtual_network.azfw_vnet.name
+  address_prefixes     = ["10.10.1.0/24"]
+  depends_on = [
+    azurerm_virtual_network.azfw_vnet
+  ]
+}
+
+// Create a Subnet for Jump VM
+resource "azurerm_subnet" "jump_subnet" {
+  name                 = "subnet-jump"
+  resource_group_name  = azurerm_resource_group.azfw_rg.name
+  virtual_network_name = azurerm_virtual_network.azfw_vnet.name
+  address_prefixes     = ["10.10.2.0/24"]
+
+  depends_on = [
+    azurerm_virtual_network.azfw_vnet,
+    azurerm_route_table.rt
```

Comment:
does the subnet need to be created after the route table? if so, remove the vnet reference and leave this one.